keystore/src/entities/platform/generic/mls/credential.rs

```diff
@@ -33,6 +33,13 @@ impl Entity for MlsCredential {
     fn id_raw(&self) -> &[u8] {
         self.id.as_slice()
     }
+
+    fn merge_key(&self) -> Vec<u8> {
+        let mut merge_key = self.id_raw().to_vec();
+        // Credentials are unique by id and type, the type is contained in these bytes.
+        merge_key.extend(self.credential.to_vec());
```

Comment:
Can we just iterate over `self.credential` in  order to avoid creating a new `Vec`, just so we can iterate over it and then immediately discard it?

Reply:
Perfect!